src/preferences/dialog/dlgprefmixer.cpp

```diff
@@ -908,28 +1007,91 @@ void DlgPrefMixer::drawXfaderDisplay() {
     graphicsViewXfader->repaint();
 }
 
-void DlgPrefMixer::slotUpdateXFader() {
-    if (radioButtonAdditive->isChecked()) {
-        m_xFaderMode = MIXXX_XFADER_ADDITIVE;
-    } else {
-        m_xFaderMode = MIXXX_XFADER_CONSTPWR;
-    }
+void DlgPrefMixer::slotXFaderReverseBoxToggled() {
+    qWarning() << "     .";
+    qWarning() << "     slotXFaderReverseChanged";
+    qWarning() << "     .";
+    m_xFaderReverse = checkBoxReverse->isChecked();
+}
 
-    // m_transform is in the range of 1 to 1000 while 50 % slider results
+void DlgPrefMixer::slotXFaderSliderChanged() {
+    qWarning() << "     .";
+    qWarning() << "     slotXFaderWidgetsChanged";
+    qWarning() << "     .";
+    // m_xFaderCurve is in the range of 1 to 1000 while 50 % slider results
     // to ~2, which represents a medium rounded fader curve.
-    double transform = RescalerUtils::linearToOneByX(
-                               SliderXFader->value(),
-                               SliderXFader->minimum(),
-                               SliderXFader->maximum(),
-                               EngineXfader::kTransformMax) -
+    double curve = RescalerUtils::linearToOneByX(
+                           SliderXFader->value(),
+                           SliderXFader->minimum(),
+                           SliderXFader->maximum(),
+                           EngineXfader::kTransformMax) -
             1 + EngineXfader::kTransformMin;
     // Round to 4 decimal places to avoid round-trip offsets with default 1.0
-    m_transform = std::round(transform * 10000) / 10000;
-    m_cal = EngineXfader::getPowerCalibration(m_transform);
+    m_xFaderCurve = std::round(curve * 10000) / 10000;
+    // If the curve has been changed in the GUI we fetch the engine value for
+    // calibration which gives us a smooth curve.
+    // This wipes any previous value set by controller mappings for example.
+    m_xFaderCal = EngineXfader::getPowerCalibration(m_xFaderCurve);
+    m_xFaderMode = radioButtonConstantPower->isChecked()
+            ? MIXXX_XFADER_CONSTPWR
+            : MIXXX_XFADER_ADDITIVE;
+
```

Comment:
does this need to be in the slider slot?

Reply:
No, good catch :+1: